controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -2004,8 +2005,24 @@ func TestInvalidBMHCanBeDeleted(t *testing.T) {
 
 	doDeleteHost(host, r)
 
+	seenDeletingState := false
+	seenDeleted := false
 	tryReconcile(t, r, host, func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
-		return host.Status.Provisioning.State == metal3v1alpha1.StateDeleting && len(host.Finalizers) == 0
```

Comment:
@asalkeld  I guess the same problem has been addressed also in #903, see here https://github.com/metal3-io/baremetal-operator/pull/903/files#diff-20c1a976622b8d54cf98349e3e6219778fa00e89ed669f1bb12cd778326469bbR142